microservices/case-service/src/main/java/org/entando/bundle/service/CaseService.java

```diff
@@ -1,14 +1,82 @@
 package org.entando.bundle.service;
 
+import org.entando.bundle.domain.CaseMetadata;
 import org.entando.bundle.entity.Process;
+import org.entando.bundle.entity.enumeration.State;
+import org.springframework.web.multipart.MultipartFile;
 
+import javax.transaction.Transactional;
 import java.util.List;
 import java.util.Optional;
 
-public abstract class CaseService {
-  public abstract List<Process> getAllProcesses();
+public interface CaseService {
 
-  public abstract Process crateProcess(Process process);
+  /**
+   * Return all processes FULL information
+   * @return the list of the processes
+   */
+  List<Process> getAllProcesses();
+
+  /**
+   * Return all processes belonging to the given user. FULL information is returned
+   * @param name the name of the owner of the process
+   * @return the list of the processes
+   */
+  List<Process> getProcessesByName(String name);
+
+  /**
+   * Persist a complete process object in the database
+   * @param process the object
+   * @return the saved process
+   */
+  Process saveProcess(Process process);
+
+  /**
+   * Return the desired process
+   * @param id process ID
+   * @return the desired process
+   */
+  Optional<Process> getProcess(Long id);
+
+  /**
+   * Return the desired process making sure that belongs to the given user
+   * @param id the process id
+   * @param name the name of the owner
+   * @return the process
+   */
+  Optional<Process> getProcessByIdAndOwner(Long id, String name);
+
+  /**
+   * Delete the process record from the database
+   * @param id the process ID
+   */
+  void deleteProcess(Long id);
+
+  /**
+   * Update the state of the desired process
+   * @param id the process ID
+   * @param state the new state
+   */
+  @Transactional
+  void updateState(Long id, State state);
```

Comment:
This is needed in order to avoid a Hibernate exception where it points out that the call must be declared transactional due to the internals of JPA / Hibernate ~ didn't dig the subject though

Reply:
The transactional is normal and usually I tag the entire service class. I meant `updateState`

Reply:
Now I understand - in case there's an error removing the resources from S3 I'm not deleting the case record from the DB  so that the operation can be retried later; the corresponding record state is marked DELETED. I already told Rax to ignore such entries but I see that there's no delete Case button in Figma for the current iteration so the method is currently useless (the same goes for the delete method).

Reply:
yes remove it